AwsCryptographyPrimitives/runtimes/go/ImplementationFromDafny-go/AESEncryption/externs.go

```diff
@@ -0,0 +1,89 @@
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm iv length %d doesn't match actual iv length %d ", algo.Dtor_ivLength(), len(ivBytes)).Error())...)))
+	}
+
+	if algo.Dtor_tagLength() != int32(len(authTagBytes)) {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm tag length %d doesn't match actual tag length %d ", algo.Dtor_tagLength(), len(authTagBytes)).Error())...)))
+	}
+
+	block, err := aes.NewCipher(keyBytes)
+	if err != nil {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
+	}
+
+	if algo.Is_AES__GCM() {
+		gcm, err := cipher.NewGCM(block)
+		if err != nil {
+			return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
+		}
+
+		plaintext, err := gcm.Open(nil, ivBytes, append(cipherTextBytes, authTagBytes...), aadBytes)
+		if err != nil {
+			return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
+		}
+		return Wrappers.Companion_Result_.Create_Success_(dafny.SeqOfBytes(plaintext))
+	}
+	return Wrappers.Companion_Result_.Create_Failure_(false)
+}
+
+func (CompanionStruct_Default___) AESEncryptExtern(algo AwsCryptographyPrimitivesTypes.AES__GCM, iv dafny.Sequence, key dafny.Sequence, msg dafny.Sequence, aad dafny.Sequence) Wrappers.Result {
+	keyBytes := dafny.ToByteArray(key)
+	ivBytes := dafny.ToByteArray(iv)
+	aadBytes := dafny.ToByteArray(aad)
+
+	if algo.Dtor_keyLength() != int32(len(keyBytes)) {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm key length %d doesn't match actual key length %d ", algo.Dtor_keyLength(), len(keyBytes)).Error())...)))
+
+	}
+
+	if algo.Dtor_ivLength() != int32(len(ivBytes)) {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(fmt.Errorf("algorithm iv length %d doesn't match actual iv length %d ", algo.Dtor_ivLength(), len(ivBytes)).Error())...)))
+	}
+
+	block, err := aes.NewCipher(keyBytes)
+	if err != nil {
+		return Wrappers.Companion_Result_.Create_Failure_(AwsCryptographyPrimitivesTypes.Companion_Error_.Create_AwsCryptographicPrimitivesError_(dafny.SeqOfChars([]dafny.Char(err.Error())...)))
+	}
+
+	if algo.Is_AES__GCM() {
```

Comment:
algo is passed in as a `algo AwsCryptographyPrimitivesTypes.AES__GCM`
why are we checking if it is and then returning false if it isnt? if it is always aes_gcm why a conditional?

Reply:
Leftover code. Removed

Reply:
hmm, perhaps a commit got stashed or lost in the latest revision?